### PR TITLE
Makes MatrixClient Closeable outside of logout function.

### PR DESCRIPTION
**Pull request by ganfra**

### CODE REVIEW

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt [hunk 1]
```diff
@@ -179,18 +179,18 @@ class RustMatrixClient constructor(
...
-        close()
         try {
             client.logout()
         } catch (failure: Throwable) {
             Timber.e(failure, "Fail to call logout on HS. Still delete local files.")
         }
         baseDirectory.deleteSessionDirectory(userID = client.userId())
         sessionStore.removeSession(client.userId())
-        client.destroy()
+        close()
```
Comment (bmarty) [inline libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt:193]: Not blocking (and not new), but is it the responsibility to the class to close itself?

**Review by bmarty — APPROVED** [review]
LGTM, thanks, just a small question.

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt (hunk 1 above)
Comment (ganfra) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClient.kt:193]: I'd say when it's a logout, yes, but both are valid options

**Comment by ganfra** [comment]
I don't understand the code quality check failure

**Comment by bmarty** [comment]
> I don't understand the code quality check failure

You can ignore for now.

**Comment by ganfra** [comment]
So I can merge?

**Pull request merged**